src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocCorrect.java

```diff
@@ -196,4 +196,120 @@ void foo6() {}
          /** Stop instances being created. **/
          String twoSentences1() {return "Sentence one. Sentence two.";}
     };
+
+    class InputInlineSummaryJavadocParagraphCheck  {
+        /**
+         *  {@summary foooo@foooo.}
+         */
+        public static final byte NUL = 0;
+
+        /**
+         * {@summary Some java@doc.
```

Comment:
is it allowed by javadoc tool to wrap `@summary` with some html tags ? like `<p>` for example
What if content will be like `brrr {@summary Some java. }`  ?

please share javadoc tool outputs in issue as comment.

Reply:
Done

Reply:
I don't think it is `@summary` can be wrapped inside an HTML Tag, although its description might be wrapped as per this line in official oracale Javadoc, **When writing the comment, make the first sentence a summary about the package, and do not put a title or any other text between the <body> tag and the first sentence.** from [javadoc](https://docs.oracle.com/javase/8/docs/technotes/tools/windows/javadoc.html)

Reply:
> please share javadoc tool outputs in issue as comment.

By issue I hope you mean #7202, or here in this PR only?

Reply:
yes, we need to keep knowledge on how javdoc tool is behaving on not simple tags content.

Reply:
For a code like 
```
/**
     * {<p>@summary No Period in end.</p>}
     */
    void foo6(){}
```
The tree is 
```
|--LEADING_ASTERISK ->      * [5:0]
|--TEXT ->  { [5:6]
|--HTML_ELEMENT -> HTML_ELEMENT [5:8]
|   `--PARAGRAPH -> PARAGRAPH [5:8]
|       |--P_TAG_START -> P_TAG_START [5:8]
|       |   |--START -> < [5:8]
|       |   |--P_HTML_TAG_NAME -> p [5:9]
|       |   `--END -> > [5:10]
|       |--TEXT -> @summary No Period in end. [5:11]
|       `--P_TAG_END -> P_TAG_END [5:37]
|           |--START -> < [5:37]
|           |--SLASH -> / [5:38]
|           |--P_HTML_TAG_NAME -> p [5:39]
|           `--END -> > [5:40]
|--TEXT -> } [5:41]
```

Reply:
Also, I think summary tag is not allowed inside any tag.

Reply:
Our AST could be any, my concern is how javadoc toilet from jdk process such cases.
If tool does not support this, Check should not also.
If tool handle javadoc comment property, Check should be aligned in behavior.

Reply:
Code - 
```
public class Test1 {

	/**
	 * Main Code.
	 * 
	 * @param args command line arguments.
	 */
	public static void main(String[] args) {
		
	}
	
	/**
     * {<p>@summary No Period in end.</p>}
     */
    public void foo1(){}
    
    /**
     * <p>{@summary Normal Javadoc.}</p>
     */
    public void foo2() {} 
}
```
Javadoc tool output :-
![image](https://user-images.githubusercontent.com/69365878/105649106-d3894b00-5ed4-11eb-87ff-4f75049cb8ee.png)

Our AST Tree
```
JAVADOC -> JAVADOC [0:0]
|--TEXT -> package test1; [0:0]
|--NEWLINE -> \r\n [0:14]
|--NEWLINE -> \r\n [1:0]
|--TEXT -> public class Test1 { [2:0]
|--NEWLINE -> \r\n [2:20]
|--NEWLINE -> \r\n [3:0]
|--TEXT -> \t/** [4:0]
|--NEWLINE -> \r\n [4:4]
|--LEADING_ASTERISK -> \t * [5:0]
|--TEXT ->  Main Code. [5:3]
|--NEWLINE -> \r\n [5:14]
|--LEADING_ASTERISK -> \t * [6:0]
|--TEXT ->   [6:3]
|--NEWLINE -> \r\n [6:4]
|--LEADING_ASTERISK -> \t * [7:0]
|--WS ->   [7:3]
|--JAVADOC_TAG -> JAVADOC_TAG [7:4]
|   |--PARAM_LITERAL -> @param [7:4]
|   |--WS ->   [7:10]
|   |--PARAMETER_NAME -> args [7:11]
|   |--WS ->   [7:15]
|   `--DESCRIPTION -> DESCRIPTION [7:16]
|       |--TEXT -> command line arguments. [7:16]
|       |--NEWLINE -> \r\n [7:39]
|       |--LEADING_ASTERISK -> \t * [8:0]
|       |--TEXT -> / [8:3]
|       |--NEWLINE -> \r\n [8:4]
|       |--TEXT -> \tpublic static void main(String[] args) { [9:0]
|       |--NEWLINE -> \r\n [9:41]
|       |--TEXT -> \t\t [10:0]
|       |--NEWLINE -> \r\n [10:2]
|       |--TEXT -> \t} [11:0]
|       |--NEWLINE -> \r\n [11:2]
|       |--TEXT -> \t [12:0]
|       |--NEWLINE -> \r\n [12:1]
|       |--TEXT -> \t/** [13:0]
|       |--NEWLINE -> \r\n [13:4]
|       |--LEADING_ASTERISK ->      * [14:0]
|       |--TEXT ->  { [14:6]
|       |--HTML_ELEMENT -> HTML_ELEMENT [14:8]
|       |   `--PARAGRAPH -> PARAGRAPH [14:8]
|       |       |--P_TAG_START -> P_TAG_START [14:8]
|       |       |   |--START -> < [14:8]
|       |       |   |--P_HTML_TAG_NAME -> p [14:9]
|       |       |   `--END -> > [14:10]
|       |       |--TEXT -> @summary No Period in end. [14:11]
|       |       `--P_TAG_END -> P_TAG_END [14:37]
|       |           |--START -> < [14:37]
|       |           |--SLASH -> / [14:38]
|       |           |--P_HTML_TAG_NAME -> p [14:39]
|       |           `--END -> > [14:40]
|       |--TEXT -> } [14:41]
|       |--NEWLINE -> \r\n [14:42]
|       |--LEADING_ASTERISK ->      * [15:0]
|       |--TEXT -> / [15:6]
|       |--NEWLINE -> \r\n [15:7]
|       |--TEXT ->     public void foo1(){} [16:0]
|       |--NEWLINE -> \r\n [16:24]
|       |--TEXT ->      [17:0]
|       |--NEWLINE -> \r\n [17:4]
|       |--TEXT ->     /** [18:0]
|       |--NEWLINE -> \r\n [18:7]
|       |--LEADING_ASTERISK ->      * [19:0]
|       |--TEXT ->   [19:6]
|       |--HTML_ELEMENT -> HTML_ELEMENT [19:7]
|       |   `--PARAGRAPH -> PARAGRAPH [19:7]
|       |       |--P_TAG_START -> P_TAG_START [19:7]
|       |       |   |--START -> < [19:7]
|       |       |   |--P_HTML_TAG_NAME -> p [19:8]
|       |       |   `--END -> > [19:9]
|       |       |--JAVADOC_INLINE_TAG -> JAVADOC_INLINE_TAG [19:10]
|       |       |   |--JAVADOC_INLINE_TAG_START -> { [19:10]
|       |       |   |--CUSTOM_NAME -> @summary [19:11]
|       |       |   |--WS ->   [19:19]
|       |       |   |--DESCRIPTION -> DESCRIPTION [19:20]
|       |       |   |   `--TEXT -> Normal Javadoc. [19:20]
|       |       |   `--JAVADOC_INLINE_TAG_END -> } [19:35]
|       |       `--P_TAG_END -> P_TAG_END [19:36]
|       |           |--START -> < [19:36]
|       |           |--SLASH -> / [19:37]
|       |           |--P_HTML_TAG_NAME -> p [19:38]
|       |           `--END -> > [19:39]
|       |--NEWLINE -> \r\n [19:40]
|       |--LEADING_ASTERISK ->      * [20:0]
|       |--TEXT -> / [20:6]
|       |--NEWLINE -> \r\n [20:7]
|       |--TEXT ->     public void foo2() {}  [21:0]
|       |--NEWLINE -> \r\n [21:26]
|       |--TEXT -> } [22:0]
|       `--NEWLINE -> \r\n [22:1]
`--EOF -> <EOF> [23:0]
```

Seems whole, {@summary } Inline tag can be

Reply:
@AkMo3 , please look at 
https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/utils/AnnotationUtil.html
images that you shared does not show what content javadoc tool considered as summary.
Summary is sentence that is moved to table above.
![image](https://user-images.githubusercontent.com/812984/108002478-a2b4a700-6fa4-11eb-82b7-4b3ffd437880.png)
so lets share only table content as is clear prove what javadoc tool consider as summary.
Please do NOT share our AST, it is details of implementation, but right now we try to define what is summary and what values we can put here.